### PR TITLE
Add ability to get the latest incremental snapshot via RPC

### PR DESCRIPTION
**Pull request by diman-io**
#### Problem

There is no ability to download latest incremental snapshot through rpc without knowing it full name with hash.

#### Summary of Changes

Added ability to download latest incremental snapshot using "/incremental-snapshot.tar.bz2" path

### CODE REVIEW

**Comment by brooksprumo** [comment]
One of the reasons I didn't add this while doing the whole Incremental Snapshots feature was because I wasn't sure how the user experience should work around this. Specifically, since an incremental snapshot (ISS) is based on a full snapshot (FSS), to download an ISS, you must know what FSS it is based on. Therefore, to ensure that the downloaded ISS is compatible with an existing FSS, you would need to have the slot and hash of the FSS.

Relaxing this a bit, I think we can get away without needing the hash of the FSS. Because if there's a hash mismatch between FSS and ISS-expected-base, the loading from the snapshots will fail. At that point downloading new snapshot(s) would likely be required. I think snapshots of the same slot with different hashes is uncommon though (gut feeling, no data to back this up!).

To download an ISS without providing the base slot, then there's the chance the ISS being download was based on a different slot. Again, this will cause the loading from snapshots to fail. So in either case, it's not a security or correctness risk, just maybe a non ideal user experience.

Downloading an incompatible ISS may cause more support requests. Maybe we need to add some error logging telling the user to manually delete this incompatible ISS? I'm not sure if we can/should do any (safe) automatic cleanup in the case.

By adding the behavior as described in this PR, we basically put the onus on the _downloader_ to ensure the _downloadee_'s ISS is based on their shared highest-full-snapshot. I'm OK with this behavior—if that's what we want—but I want to make sure all UX issues are known upfront.

@mvines wdyt?

File: rpc/src/rpc_service.rs [hunk 1]
```diff
@@ -227,16 +227,37 @@ impl RequestMiddleware for RpcRequestMiddleware {
         trace!("request uri: {}", request.uri());
 
         if let Some(ref snapshot_config) = self.snapshot_config {
-            if request.uri().path() == "/snapshot.tar.bz2" {
+            if request.uri().path() == "/snapshot.tar.bz2"
+                || request.uri().path() == "/incremental-snapshot.tar.bz2"
```
Comment (brooksprumo) [inline rpc/src/rpc_service.rs:231]: Can you turn these into constants? How about:

```rust
const FULL_SNAPSHOT_REQUEST_PATH: &str = "/snapshot.tar.bz2";
const INCREMENTAL_SNAPSHOT_REQUEST_PATH: &str = "/incremental-snapshot.tar.bz2";
```

**Comment by mvines** [comment]
> By adding the behavior as described in this PR, we basically put the onus on the downloader to ensure the downloadee's ISS is based on their shared highest-full-snapshot. I'm OK with this behavior—if that's what we want—but I want to make sure all UX issues are known upfront.

I think this is fine.  What we have today is the ability to manually download the latest full snapshot from a given `$host` with this shortcut:
```
$ wget --trust-server-names http://$host/snapshot.tar.bz2
```

Extending this to include the latest incremental makes perfect sense to me:
```
$ wget --trust-server-names http://$host/incremental-snapshot.tar.bz2
````

The user certainly needs to be aware that the incremental they just downloaded doesn't match the full that they currently have, and then deal with it.  `solana-validator init` can be used to download an incremental and/or full snapshot as well, but the quick `wget` option for advanced users is nice too

File: rpc/src/rpc_service.rs (hunk 1 above)
Comment (diman-io) [reply rpc/src/rpc_service.rs]: Done

**Review by brooksprumo — APPROVED** [review]

**Comment by brooksprumo** [comment]
@mvines Do you want me to do the merge and watch/manage the backport?

**Comment by mvines** [comment]
opps, we needed 99f1a43262bda1f234d645449ddb93ca3c2ae4be for the v1.10 backport label to work

**Comment by brooksprumo** [comment]
@Mergifyio backport v1.10

**Pull request merged**